plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/ConfigSetup.java

```diff
@@ -0,0 +1,162 @@
+package io.jenkins.tools.pluginmanager.cli;
+
+import io.jenkins.tools.pluginmanager.config.Config;
+import io.jenkins.tools.pluginmanager.config.Settings;
+import io.jenkins.tools.pluginmanager.impl.Plugin;
+import java.io.FileNotFoundException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import org.apache.commons.lang3.StringUtils;
+
+
+public class ConfigSetup {
+    List<Plugin> plugins;
+    Config cfg;
+    CliOptions options;
+
+    public ConfigSetup(Config cfg, CliOptions options) {
```

Comment:
Alternatively, instead of taking in a `Config` since that's going to be empty every time, the `setup()` method could return a completed `Config` and only have to take in whatever is coming in from the `CliOptions`